src/main.c

```diff
@@ -161,6 +161,12 @@ __libspec int main(int argc, char *argv[])
   }
 #endif // __APPLE__
 
+#ifdef ANDROID
+  // Accept argv[1] passed in from the Java side as the "intended" argv[0].
+  if(argc >= 2)
+    argv[0] = argv[1];
+#endif
```

Comment:
If there's a way any other args might get passed here it might be better to shift argv by one so it doesn't think the executable is also supposed to be the startup path (see line 229).

Reply:
We control the arguments fully Java-side (except arg0), but this is a fair argument - and would probably be just the tiny bit cleaner.